docs/tutorial/index.md

```diff
@@ -1,609 +1,553 @@
+- On the MicroK8s side (in MicroK8s' database) you've now got:
+  1. a 'controller' namespace (`kubectl get namespaces`) holding
+  1. a 'controller-0' pod (`kubectl -n controller-my-first-juju-controller get pods`) holding
+  1. 3 containers: 'api-server', 'charm', 'mongodb' (`kubectl -n controller describe pod controller-0`).
+- On the Juju side (in the Juju controller's database) you've now got:
+  1. a Juju 'my-first-juju-controller' controller (i.e., control plane), with access to the 'microk8s' cloud (`juju show-controller my-first-juju-controller`), and holding
+  1. a Juju model (i.e., workspace; corresponding to a Kubernetes namespace) implicitly called 'controller', using resources from the 'microk8s' cloud (`juju switch controller`, then `juju show-model controller`), and  holding
+  1. a Juju application (i.e., deployed charm -- in this case, the `juju-controller` charm) called 'controller' with a Juju unit (i.e., replica; corresponding to a Kubernetes pod) called 'controller/0', deployed on a resource from the 'microk8s' cloud (`juju status`), and holding
+  1. the same 3 containers we saw on the MicroK8s side (`juju ssh --container [api-server | charm | mongodb] 0`), which hold the various bits of a running Juju controller:
+    - 'charm': contains a Juju unit agent and `juju-controller` charm code. The Juju unit agent pings the Juju controler agent (see next item) to check the unit's target state against the state in the database (see two items down) and progresses the unit's state by executing the charm code.
+    - 'api-server': contains Pebble, a lightweight process supervisor, and the Juju controller agent, which also wraps the Juju API server. The controller agent reads and writes to the database (see next item).
+    - 'mongodb': contains Juju's internal database.
 
-> See more:
-> - [Charmhub | `postgresql-k8s` > Actions > `set-password`](https://charmhub.io/postgresql-k8s/actions#set-password) (or run `juju actions postgresql-k8s` followed by `juju show-action postgresql-k8s set-password`)
-> - {ref}`manage-actions`
+These details are specific to Kubernetes, bootstrap, and the controller application, but this is fundamentally how Juju operates for all clouds, all operations, and all applications: the cloud provides a resource and Charmhub provides charms, and the result is a resource that holds the Juju unit agent and the charm code, where the Juju unit agent always pings the Juju controller agent to reconcile state, then executes the charm code accordingly, and the charm reacts as it's been programmed to react, e.g., by installing a workload.
 
 
+Back to our main concern, to be fully operational a controller needs access to a cloud and to Charmhub. We saw just now that our controller already has access to our 'microk8s' cloud -- this access was granted implicitly through bootstrap. Also, as before, so long as you're connected to the internet, your controller has access to Charmhub too. Your controller is all set!
```

Comment:
> to be fully operational a controller needs access to a cloud and to Charmhub.

That's only true for this tutorial. You can have air gapped setups that doesn't require Charmhub at all i.e. you can make charms without ever using Charmhub i.e. you can bootstrap with a local charm.

```
juju bootstrap lxd test --controller-charm-path=./juju-controller_r117.charm
```